src/realm/array_encode.hpp

```diff
@@ -40,8 +40,18 @@ class ArrayEncode {
     // init from mem B
     inline size_t size() const;
     inline size_t width() const;
-    inline uint64_t width_mask() const;
+    inline size_t ndx_size() const;
```

Comment:
I'm not sure why we would want to hide the fields? this introduces a lot of boilerplate code.

Reply:
I got rid of most of them, but in general some minimal getter is still needed, I don't like to have `ArrayPacked` and `ArrayFlex` friends of `ArrayEncode`